eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -3224,6 +3224,84 @@ public function testMoveSubtreeUpdatesPathIdentificationString(): void
         );
     }
 
+    /**
+     * Test that is_visible is set correct for children when moving a content which is hidden (location is not hidden).
+     *
+     * @covers \eZ\Publish\API\Repository\LocationService::moveSubtree
+     * @depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testLoadLocation
+     */
+    public function testMoveSubtreeKeepsContentHidden(): void
+    {
+        $repository = $this->getRepository();
+
+        $mediaLocationId = $this->generateId('location', 43);
+        $demoDesignLocationId = $this->generateId('location', 56);
+        /* BEGIN: Use Case */
+        // $mediaLocationId is the ID of the "Media" page location in
+        // an eZ Publish demo installation
+
+        // $demoDesignLocationId is the ID of the "Demo Design" page location in an eZ
+        // Publish demo installation
+
+        // Load the location service
+        $locationService = $repository->getLocationService();
+
+        // Load the content service
+        $contentService = $repository->getContentService();
+
+        // Load location to move
+        $locationToMove = $locationService->loadLocation($demoDesignLocationId);
+
+        // Create child below locationToMove
+        $subFolderContent = $this->publishContentWithParentLocation('SubFolder', $locationToMove->id);
+
+        // Hide source Content
+        $contentService->hideContent($locationToMove->contentInfo);
+        $locationToMove = $locationService->loadLocation($demoDesignLocationId);
+
+        // Load new parent location
+        $newParentLocation = $locationService->loadLocation($mediaLocationId);
+
+        // Move location from "Home" to "Media"
+        $locationService->moveSubtree(
+            $locationToMove,
+            $newParentLocation
+        );
+
+        // Load moved location
+        $movedLocation = $locationService->loadLocation($demoDesignLocationId);
+        /* END: Use Case */
+
+        // Assert Moved Location
+        $this->assertPropertiesCorrect(
+            [
+                'hidden' => true, // It should be hidden only on object level, not on location level. But impossible to say due to https://github.com/ezsystems/ezpublish-kernel/blob/7.5/eZ/Publish/Core/Repository/Helper/DomainMapper.php#L562
```

Comment:
AFAICS after this assert you're doing
```php
self::assertTrue($movedLocation->getContentInfo()->isHidden);
```
which is the answer to some of those doubts. We still won't know if Location is hidden, but is this really a subject of the test? The test checks if content is hidden after a subtree move, so IMHO it's enough.

Reply:
The problem with [EZP-30746](https://github.com/ezsystems/ezpublish-kernel/pull/3142#) is that hidden status for objects became wrong. So I think it would be good if this test could ensure everything are set correctly.

I would like to suggest that we add the possibility to see via API why a location is hidden. I think this could be done fairly easy properties Location, and setting them in `DomainMapper::mapLocation()`:
- isHiddenByLocation
- isHiddenByContent
- isHiddenByParent

I could do this in this PR, or in a separate one of you think it is an good idea